src/main/java/org/isf/usersettings/rest/UserSettingController.java

```
@@ -0,0 +1,203 @@
+		return userSettingsDTO;
+	}
+
+	/**
+	 * Returns a {@link UserSettingDTO} of userSetting created.
+	 * @param userSettingDTO -  the {@link UserSettingDTO} to insert.
+	 * @return {@link UserSettingDTO} if the userSetting has been created, null otherwise.
+	 * @throws OHServiceException
+	 */
+	@ResponseStatus(HttpStatus.CREATED)
+	@PostMapping(value = "/usersettings", produces = MediaType.APPLICATION_JSON_VALUE)
+	public UserSettingDTO newUserSettings(@Valid @RequestBody UserSettingDTO userSettingDTO) throws OHServiceException {
+		LOGGER.info("Create a UserSetting.");
+		String requestUserName = userSettingDTO.getUser();
+		String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();
+		if (!requestUserName.equals(currentUser)) {
+			throw new OHAPIException(new OHExceptionMessage("Not allowed."), HttpStatus.FORBIDDEN);
+		}
+		if (userSettingManager.getUserSettingByUserNameConfigName(requestUserName, userSettingDTO.getConfigName()) != null) {
+			throw new OHAPIException(new OHExceptionMessage("A setting with that name already exists."));
+		}
+		if (userManager.getUserByName(requestUserName) == null) {
+			throw new OHAPIException(new OHExceptionMessage("The specified user does not exist."));
+		}
+		userSettingDTO.setId(0);
+		UserSetting userSetting = userSettingMapper.map2Model(userSettingDTO);
+		UserSetting created = userSettingManager.newUserSetting(userSetting);
+		if (created == null) {
+			LOGGER.info("UserSetting is not created.");
+			throw new OHAPIException(new OHExceptionMessage("UserSetting not created."));
+		}
+		LOGGER.info("UserSetting successfully created.");
+		return userSettingMapper.map2DTO(created);
+	}
+
+	/**
+	 * Updates an existing {@link UserSettingDTO}.
+	 * @param userSettingDTO - the {@link UserSettingDTO} to update.
+	 * @param id - id of {@link UserSetting} .
+	 * @return {@link UserSettingDTO} if the UserSetting has been updated , null otherwise.
```

The code implies that an exception is thrown if the settings are not updated instead of returning `null`